core/templates/dev/head/components/ExplorationSummaryTileDirective.js

```diff
@@ -78,7 +82,48 @@ oppia.directive('explorationSummaryTile', [function() {
           }
         );
 
-        $scope.MAX_CONTRIBUTORS_TO_DISPLAY = 5;
+        $scope.avatarsList = [];
+        $scope.contributors.forEach(function(contributorName) {
+          var DEFAULT_PROFILE_IMAGE_PATH = '/images/avatar/user_blue_72px.png';
+
+          var avatarData = {
+            image: DEFAULT_PROFILE_IMAGE_PATH,
+            tooltipText: contributorName
+          };
+
+          var profileImageUrl = (
+            '/preferenceshandler/profile_picture_by_username/' +
+              contributorName);
+          // Returns a promise for the user profile picture, or
+          // the default image if user is not logged in or
+          // has not uploaded a profile picture, or
+          // the player is in preview mode.
+          $http.get(profileImageUrl).success(function(data) {
```

Comment:
It seems really wasteful to do so many $http.gets here (in fact, this gets multiplied by the number of summary tiles on the page). Instead, would it be possible to modify the backend so that data about the contributors' profile image data url gets passed along to the frontend and no further get() calls are needed? Now that this code is here, as opposed to being in various isolated profile-link-image directives, it seems like a very obvious thing to do.

The relevant code is in core/domain/summary_services.py -- I suggest modifying the 'contributors summary' field and turning it into something like

```
{
  username: {
    num_commits: ...,
    profile_picture_data_url: ...,
  }
}
```

Please be careful to modify the clients of the contributors summary accordingly, though. I also noted that there's a 'contributor_names' field which seems to contain information that duplicates the human_readable_contributors_summary; it might be worth getting rid of that field (but please check if anything else uses it first).